saged/disease_vector.py

```diff
@@ -0,0 +1,170 @@
+    # Set tf to be deterministic
+    session_conf = tf.ConfigProto(
+        intra_op_parallelism_threads=1, inter_op_parallelism_threads=1
+    )
+
+    # Create tf session
+    sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
+    K.set_session(sess)
+
+    # Train a VAE on the data
+    expression_array, _ = expression_data.get_all_data()
+    expression_array = labeled_scaler.transform(expression_array)
+
+    sample_ids = expression_data.get_samples()
+
+    metadata['sample_ids'] = sample_ids
+
+    expression_df = pd.DataFrame(expression_array, index=sample_ids)
+
+    sample_to_label = expression_data.sample_to_label
+
+    samples = expression_df.index
+    labels = samples.map(sample_to_label)
+
+    disease_df = expression_df[labels.str.match(args.label)]
+    healthy_df = expression_df[labels.str.match(args.negative_class)]
+
+    expression_encoder, expression_decoder, hist = vae.run_tybalt_training(expression_df,
+                                                                           learning_rate,
+                                                                           batch_size,
+                                                                           epochs,
+                                                                           kappa,
+                                                                           intermediate_dim,
+                                                                           latent_dim,
+                                                                           epsilon_std,
+                                                                           val_frac,)
+
+    # Calculate the centroids for the embeddings of the disease and healthy samples
+    disease_embeddings = expression_encoder.predict_on_batch(disease_df)
+    healthy_embeddings = expression_encoder.predict_on_batch(healthy_df)
+
+    disease_centroid = disease_embeddings.mean(axis=0)
+    healthy_centroid = healthy_embeddings.mean(axis=0)
+    disease_std = disease_embeddings.std(axis=0)
+    healthy_std = healthy_embeddings.std(axis=0)
+
+    # Sample disease data
+    full_sample = np.zeros((args.sample_count, disease_embeddings.shape[1]))
+    for i in range(args.sample_count):
+        full_sample[i, :] = np.random.normal(loc=disease_centroid,
+                                             scale=disease_std)
+
+    sim_disease = expression_decoder.predict_on_batch(full_sample)
+
+    out_path = os.path.join(args.out_dir, 'joint_sim_disease.pkl')
+    with open(out_path, 'wb') as out_file:
+        pickle.dump(sim_disease.T, out_file)
+
+    # TODO save result
+
+    disease_healthy_difference = disease_centroid - healthy_centroid
+    for interpolation_amount in range(0, 11, 2):
+        # Calculate the shifted centroid
+        sim_centroid = healthy_centroid + (interpolation_amount * .1 * disease_healthy_difference)
+
+        # Is there a good way to interpolate the standard deviation for each LV?
```

Comment:
Have you looked at how much the healthy vs. diseased standard deviations vary in the latent space? In the UMAP plots the data variances look pretty similar, so maybe you don't need to interpolate at all.

Reply:
I agree, the simulation looks pretty good. The standard deviations do differ between the classes, but apparently not enough to worry about. I think I'll stick with not interpolating